src/Sql/Sql.Test/ScenarioTests/ManagedInstanceActiveDirectoryAdministratorTests.ps1

```diff
@@ -0,0 +1,79 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+	.SYNOPSIS
+	Tests for managing Active Directory Administrator on managed instance
+#>
+function Test-ManagedInstanceActiveDirectoryAdministrator
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$vnetName = "cl_initial"
+	$subnetName = "Cool"
+
+	# Setup VNET 
+	$virtualNetwork1 = CreateAndGetVirtualNetworkForManagedInstance $vnetName $subnetName $rg.Location
+	$subnetId = $virtualNetwork1.Subnets.where({ $_.Name -eq $subnetName })[0].Id
+
+	$managedInstance = Create-ManagedInstanceForTest $rg $subnetId
+
+	$activeDirectoryGroup1 = "aadadmin"
```

Comment:
Can't you create these as part of the test?  What work will the next person who re-records the test have to do to re-record this test?  The answer should be 'none'

Reply:
I am aware that this is a problem for someone who tried to re-record the test again. If these hardcoded users are not present in Active Directory, Set-AzSqlInstanceActiveDirectoryAdministrator command will fail with error that there is  no user/group for provided DisplayName and ObjectId. So, solution is to use existing users from provided ActiveDirectory.
I've tried to solve this by creating new AD group and user on beginning of the test, and remove them in finally section. Commands for manipulation with AAD are defined in Az.Resource module, and there is a problem to include this module into Sql. Here is the error: "Package 'Microsoft.Azure.Management.Resources 2.
20.1-preview' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Ve
rsion=v2.0'. This package may not be fully compatible with your project"

Also, there is another problem here, I've tried to execute ActiveDirectoryTests from Resource project in record mode and all tests failed with error **Operation returned an invalid status code 'Forbidden'**. I've follow this document https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/azure-powershell-developer-guide.md#ad-scenario-tests to setup AD Scenario Tests but it doesn't help.

Reply:
Because of the two problems described above, I've updated test with appropriate comment. So if someone try to re-record this test again, he will be aware that these values had to be changed.